pkg/collector/parser/receiver_jaeger.go

```diff
@@ -82,9 +87,13 @@ func (j *JaegerReceiverParser) Ports() ([]corev1.ServicePort, error) {
 			// if not, we use the default port
 			if protocolPort == nil {
 				protocolPort = &corev1.ServicePort{
-					Name: portName(nameWithProtocol, protocol.defaultPort),
-					Port: protocol.defaultPort,
+					Name:     portName(nameWithProtocol, protocol.defaultPort),
+					Port:     protocol.defaultPort,
+					Protocol: protocol.protocol,
 				}
+			} else {
+				// port from configuration block has been used, but protocol needs to be set
+				protocolPort.Protocol = protocol.protocol
```

Comment:
You could simplify this code by setting the protocol after this conditional. The current code would then be the same, but you'd just set the protocol before appending it to the list of ports.

Reply:
Thanks for spotting this @jpkrohling! I've broken this out of the conditional so that the port's protocol is now consistently set regardless of if a port is specified in the supplied configuration or not.